### ISSUE TITLE
Creating a single binary file with embedded assets

### ISSUE THREAD

**Issue by jsegura** [revel/revel#589]
I need some advice for the implementation/deployment of a revel app. The app basically serves some static files applying some logic. One of the requirements of the project is that all should be server from a single binary, this means that I should include all assets inside that binary. (no src nor media folder is allowed). This also means that the config should be embedded in the binary too.

I've been investigating for a while and I haven't found a way to do it with the legacy revel framework. I've seen some solutions like go-bindata[0] but I don't really found a way to do it without modifying the revel source code.

Have anyone worked in an environment like that?

**Comment by pedromorgan** [comment]
So i need for my next project to create a single executable, images, css et all ... in one..(idea is to run on Raspberry Pi's in a lab, around 20 of them as simple info screens)

I notice that `revel package my/app` creates an exectable, and also the whole source directory.. and revel directories....

Would go-bindata solve this problem ??

**Comment by brendensoares** [comment]
@pedromorgan no, it's not that easy. There are some code dependencies in Revel with the source for controllers and routes. I am interested in single binary builds still though, but it may result in a major shift in Revel's design under the hood.
